tools/renderer_html.py

```
@@ -0,0 +1,582 @@
+                # Take screenshot of viewport only (not full page)
+                page.screenshot(path=str(out_path), full_page=False)
+                
+                render_time = (time.time() - start_time) * 1000
+                logger.info(f"Rendered PNG: {out_path} ({render_time:.0f}ms)")
+                
+            finally:
+                # Clean up temp file
+                Path(temp_html_path).unlink(missing_ok=True)
+                browser.close()
+                
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)